spec/rubocop/cop/sorbet/refinement_spec.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Sorbet::Refinement, :config do
+  it 'registers an offense for using refine' do
+    expect_offense(<<-RUBY)
+      module Foo
+        refine(Date) do
+        ^^^^^^^^^^^^ Do not use Ruby Refinements library as it is not supported by Sorbet.
+        end
+      end
+    RUBY
+  end
+
+  it 'registers an offense for using a refinement' do
+    expect_offense(<<-RUBY)
+      module Foo
+        using(Date)
+        ^^^^^^^^^^^ Do not use Ruby Refinements library as it is not supported by Sorbet.
+      end
+    RUBY
+  end
+
+  it 'registers no offense for using non-refinement method #refine' do
+    expect_no_offenses(<<-RUBY)
+      module Foo
+        bar.refine(Date)
+      end
+    RUBY
+  end
+
+  it 'registers no offense for using non-refinement method #using' do
+    expect_no_offenses(<<-RUBY)
+      module Foo
+        bar.using(Date)
+      end
+    RUBY
+  end
+
+  it "reports an offense for use of using" do
+    expect_offense(<<~RUBY, "my_class.rb")
+      using MyRefinement
+      ^^^^^^^^^^^^^^^^^^ Do not use Ruby Refinements library as it is not supported by Sorbet.
+    RUBY
+  end
+
+  it "reports an offense for use of refine" do
+    expect_offense(<<~RUBY, "my_refinement.rb")
+      module MyRefinement
+        refine(String) do
+        ^^^^^^^^^^^^^^ Do not use Ruby Refinements library as it is not supported by Sorbet.
+          def to_s
+            "foo"
+          end
+        end
+      end
+    RUBY
+  end
+
+  it "reports no offense for use of using with non-const argument" do
+    expect_no_offenses(<<~RUBY, "my_class.rb")
+      using "foo"
+    RUBY
+  end
+
+  it "reports no offense for use of refine with non-const argument" do
+    expect_no_offenses(<<~RUBY, "my_refinement.rb")
+      module MyRefinement
+        refine "foo" do
+          def to_s
+            "foo"
+          end
+        end
+      end
+    RUBY
+  end
+
+  it "reports no offense for use of refine with no block argument" do
+    expect_no_offenses(<<~RUBY, "my_refinement.rb")
+      module MyRefinement
+        refine(String)
+      end
+    RUBY
+  end
+
+  it "reports no offense for use of refine outside of module" do
+    expect_no_offenses(<<~RUBY, "my_refinement.rb")
+      # module MyNamespace
```

Comment:
This safeguard is inadequate and this spec is not met:

`return unless node.each_ancestor(:module).any?`